source/agora/consensus/protocol/Nominator.d

```diff
@@ -451,6 +453,16 @@ extern(D):
             return;  // slot was already externalized, ignore outdated message
         }
 
+        const challenge = SCPStatementHash(&envelope.statement).hashFull();
+        PublicKey key = PublicKey(envelope.statement.nodeID);
+        Point K = Point(key[]);
+
+        if (!verify(K, envelope.signature, challenge))
+        {
+            log.trace("Envelope signature is invalid for {}: {}", key, envelope);
```

Comment:
```suggestion
            log.trace("Envelope signature is invalid for {}: {}", key, scpPrettify(&envelope));
```
This will make it easier to read.

Reply:
Thanks I updated.